ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,145 @@ def test_trustdomain_disable_disables_subdomain(self):
                 assert 'no such user' in res.stderr_text
         # verify the user can be retrieved after re-enabling trustdomain
         self.master.run_command(['id', user])
+
+
+class TestNestedMembers(IntegrationTest):
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        tasks.install_client(cls.master, cls.clients[0])
+
+    def test_nested_group_members(self):
+        """Nested group memberships should be honoured
+
+        "c" should be a child of "b"
+        so that parent child relationship is as follows:
+        "a"->"b"->"c"
+
+        user1 is direct member of "a" and as a result
+        member of "b" and "c"".
+        Now if one adds a direct membership to child1
+        nothing will change.
+
+        Now if one removes the direct membership to "b"
+        nothing should change, the memberships should be honored
+        Linked Issue: https://pagure.io/SSSD/sssd/issue/3636
+        """
+        master = self.master
+        client = self.clients[0]
+        tasks.kinit_admin(master)
+
+        # add hbacrule
+        master.run_command(['ipa', 'hbacrule-add', '--usercat', 'all',
+                            '--hostcat', 'all', '--servicecat', 'all',
+                            'any_to_any'])
+        # add a user
+        username = "testuser" + str(random.randint(200000, 9999999))
+        userpasswd = 'Secret123'
+        cmd = ['ipa', 'user-add',
+               '--first', username,
+               '--last', username,
+               '--password', username]
+        input_passwd = '{upasswd}\n{upasswd}\n'.format(upasswd=userpasswd)
+        cmd_output = master.run_command(cmd)
+        assert 'Added user "%s"' % username in cmd_output.stdout_text
+
+        cmd_output = master.run_command(['ipa', 'passwd', username],
+                                        stdin_text=input_passwd)
+        assert 'Changed password' in cmd_output.stdout_text
+
+        # add group a
+        cmd_output = master.run_command(['ipa', 'group-add', 'a'])
+        assert 'Added group "child1"' in cmd_output.stdout_text
+
+        # add group b
+        cmd_output = master.run_command(['ipa', 'group-add', 'b'])
+        assert 'Added group "child2"' in cmd_output.stdout_text
+
+        # add group parent
+        cmd_output = master.run_command(['ipa', 'group-add', 'c'])
+        assert 'Added group "parent"' in cmd_output.stdout_text
```

Comment:
Adding group `c` would result in `Added group "parent"` ?

Reply:
Another typo. I had earlier used parent, child1 and child2 but later changed it to a, b, c.
Will fix this